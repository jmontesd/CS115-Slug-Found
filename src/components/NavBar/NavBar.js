import React from 'react';
import { Link } from 'react-router-dom';
import './NavBar.scss';
import { connect } from 'react-redux';
import { signOut as actionSignOut } from '../../store/actions/authActions';

const NavBar = (props) => {
  const { isSignedIn, signOut } = props;

  const renderLogInSignUpButtons = (
    <div className="d-flex flex-row">
      <Link to="/login" className="btn btn-outline-primary mr-2">
        Log In
      </Link>
      <Link to="/signup" className="btn btn-primary">
        Sign Up
      </Link>
    </div>
  );

  const renderSignOutButton = (
    <div className="d-flex flex-row">
      <Link className="btn btn btn-outline-primary" to="/" onClick={signOut}>
        Sign Out
      </Link>
    </div>
  );

  return (
    <nav className="navbar navbar-light bg-light mb-2">
      <div className="container d-flex justify-content-between">
        <Link className="navbar-brand" to="/">
          {'Slug&Found'}
        </Link>
        {isSignedIn ? renderSignOutButton : renderLogInSignUpButtons}
      </div>
    </nav>
  );
};

const mapDispatchToProps = (dispatch) => ({
  signOut: () => dispatch(actionSignOut()),
});
const mapStateToProps = (state) => ({
  isSignedIn: state.firebase.auth.uid,
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(NavBar);
