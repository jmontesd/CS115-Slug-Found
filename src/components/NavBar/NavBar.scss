nav{
    transform: translateX(220px);  
    width: 90%;
}