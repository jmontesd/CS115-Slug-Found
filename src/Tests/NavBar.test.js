import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { NavBar } from '../components/NavBar/NavBar';

// needed for shallow rendering
configure({ adapter: new Adapter() });

describe('<NavBar />', () => {
  it('should NavBar properly when logged in', () => {
    // test creates snapshot of component and checks if matches
    // previously create snapshot
    const wrapper = shallow(<NavBar isLoggedIn />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should NavBar properly when not logged in', () => {
    // test creates snapshot of component and checks if matches
    // previously create snapshot
    const wrapper = shallow(<NavBar />);
    expect(wrapper).toMatchSnapshot();
  });
});
