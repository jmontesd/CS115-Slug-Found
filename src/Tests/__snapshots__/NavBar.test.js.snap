// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should NavBar properly when logged in 1`] = `ShallowWrapper {}`;

exports[`<NavBar /> should NavBar properly when not logged in 1`] = `ShallowWrapper {}`;
